pkg/tools/prefix_pool/prefixpool_test.go

```diff
@@ -0,0 +1,413 @@
+package prefix_pool
```

Comment:
```suggestion
package prefix_pool_test
```

Reply:
Should we really create another package for test file?

Reply:
Yes, we are not using white-box testing.

Reply:
In this case we need to export some functions from `prefixpool` package, is it ok?

Reply:
I think we need just remove white-box tests and add more tests for the prefix pool.